### PR TITLE
Frontend refactoring

### PR DESCRIPTION
**Pull request by Adebesin-Cell**
## Frontend refactoring

- Created ``generateTooltip`` component to extract duplicate code in the ``CardActions``
- Removed red bg on close button.
- Bumped node version from 16 to 18.

### CODE REVIEW

**Review by Tolu-Mals — APPROVED** [review]
Looks good. Nicely done ✨

File: src/components/Search/AnswerCard/CardActions.tsx [hunk 1]
```diff
@@ -53,90 +54,17 @@ const CardActions = ({ onShareOpen, query, answer }: CardActionsProps) => {
...
+      {generateTooltip(
+        "Like",
+        feedbackAction === "liked" ? RiThumbUpFill : RiThumbUpLine,
+        handleLike,
+      )}
+      {generateTooltip(
+        "Dislike",
+        feedbackAction === "disliked" ? RiThumbDownFill : RiThumbDownLine,
+        handleDislike,
+      )}
```
Comment (Royal-lobster) [inline src/components/Search/AnswerCard/CardActions.tsx:67]: Nice abstraction! But it's an anti pattern to return elements with a function call, instead we can make this a react component and pass function arguments as props

Comment (Adebesin-Cell) [reply src/components/Search/AnswerCard/CardActions.tsx:67]: Okay, noted. I'll fix it now. Thx! 👍

File: src/components/Search/AnswerCard/CardActions.tsx [hunk 2]
```diff
@@ -54,17 +54,21 @@ const CardActions = ({ onShareOpen, query, answer }: CardActionsProps) => {
...
-        "Like",
-        feedbackAction === "liked" ? RiThumbUpFill : RiThumbUpLine,
-        handleLike,
-      )}
-      {generateTooltip(
-        "Dislike",
-        feedbackAction === "disliked" ? RiThumbDownFill : RiThumbDownLine,
-        handleDislike,
-      )}
+      <GenerateTooltip
```
Comment (Tolu-Mals) [inline src/components/Search/AnswerCard/CardActions.tsx:57]: I think it's better to rename as `TooltipButton` since it's a component now

Comment (Adebesin-Cell) [reply src/components/Search/AnswerCard/CardActions.tsx:57]: Ha, okay 👍

**Review by Royal-lobster — APPROVED** [review]

**Pull request merged**